From 7dbf7f90bc6708f049648eb2f88cf8699d5df590 Mon Sep 17 00:00:00 2001
From: Martin Lang <martin.lang@mpsd.mpg.de>
Date: Wed, 19 Nov 2025 11:50:25 +0100
Subject: [PATCH 6/7] Add size argument to transfer calls

GCC and Flang (on Windows) seem to interpret the standard differently and flang fails.

https://github.com/UppASD/UppASD/issues/102
---
 source/Multiscale/formats.f90 | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/source/Multiscale/formats.f90 b/source/Multiscale/formats.f90
index 3f71fb4..0fbead5 100644
--- a/source/Multiscale/formats.f90
+++ b/source/Multiscale/formats.f90
@@ -376,28 +376,28 @@ contains
     type(genType) :: gt
     gt%type = TYPE_INTEGER
     gt%data = 0
-    gt%data = transfer(v,gt%data)
+    gt%data = transfer(v,gt%data,size(gt%data))
   end function gen_type_integer
   function gen_type_real_sp(v) result(gt)
     implicit none
     real, intent(in) :: v
     type(genType) :: gt
     gt%type = TYPE_REAL_SP
-    gt%data = transfer(v,gt%data)
+    gt%data = transfer(v,gt%data,size(gt%data))
   end function gen_type_real_sp
   function gen_type_real_dp(v) result(gt)
     implicit none
     real(dblprec), intent(in) :: v
     type(genType) :: gt
     gt%type = TYPE_REAL_DP
-    gt%data = transfer(v,gt%data)
+    gt%data = transfer(v,gt%data,size(gt%data))
   end function gen_type_real_dp
   function gen_type_logical(v) result(gt)
     implicit none
     logical, intent(in) :: v
     type(genType) :: gt
     gt%type = TYPE_LOGICAL
-    gt%data = transfer(v,gt%data)
+    gt%data = transfer(v,gt%data,size(gt%data))
   end function gen_type_logical
   function gen_type_str(v) result(gt)
     implicit none
@@ -408,7 +408,7 @@ contains
        stop "String too large for gen_type"
     end if
     gt%data = ichar(' ')
-    gt%data = transfer(v,gt%data)
+    gt%data = transfer(v,gt%data,size(gt%data))
   end function gen_type_str
 
 
